unified-codes/apps/data-service/src/app/schema.ts

```diff
@@ -19,7 +19,13 @@ const typeDefs = gql`
     "Request an entity by code"
     entity(code: String!): Entity
     "Request all entities with optional filter - Default behaviour: return all medicinal_products"
-    entities(filter: SearchFilter): [Entity]
+    entities(filter: SearchFilter, first: Int, offset: Int): EntitiesResponse
+  }
+
+  type EntitiesResponse {
+    data: [Entity]!
+    hasMore: Boolean!
+    totalResults: Int!
```

Comment:
Update: I think the above is misleading as it reads like you're not using repeated requests for each "batch" (which I'm aware you are). what I mean't to say is that it may be better for the `hasMore` and `totalResults` properties to be handled using a separate query. I think I explained it better in [this comment](https://github.com/openmsupply/unified-codes/pull/169#discussion_r484826918).

Reply:
Update: I think the above is misleading as it reads like you're not using repeated requests for each "batch"... I should have specified that I;m referring to the `hasMore` and `totalResults` properties as being handled using a separate query. I think this is all a duplicate of what I explain better in [this comment](https://github.com/openmsupply/unified-codes/pull/169#discussion_r484826918).

Reply:
Update: I think previous comment here was incorrect/confusing, so have deleted! Ignore!